### PR TITLE
[script.audio.profiles] 1.4.2

### PR DESCRIPTION
**Pull request by pkscout**
### Description

I've agreed to take on this abandoned addon. The updates are mostly for Kodi 18/19 comatability and fix a few bugs. I ran this through a Codacy review and also fixed a few things I knew would probably cause problems at review, but I won't be suprised if the corrections list is long for this.

v.1.4.2
- added option to disable auto load of profile at startup if it's the one that was enabled at shutdown
- added option to set delay before pop up dialog automatically closes
- added option to enable debug logging
- fix for keymap option to pop up dialog not working
- fix so the auto loaded profile for startup is always loaded
- fix so setting to pop up dialog box when playback start works again
- fix so save notifications show even if notifications are off
- fix so you don't have to restart Kodi for settings changes to be reflected
- fix for crash when trying to toggle profiles
- updated language files
- cleaned up some setting descriptions
- changes for better code compatibility between Kodi 18 and 19
- changes in logging to accomodate new logging requirements for Kodi 19

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by pkscout** [comment]
Let me know if I need to resubmit this as one PR. It took me a couple tries to get it through Travis Buddy.

File: script.audio.profiles/addon.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="script.audio.profiles" name="Audio Profiles" version="1.4.2~beta5" provider-name="pkscout, notoco, Regss">
```
Comment (enen92) [inline script.audio.profiles/addon.xml:2]: beta?

File: script.audio.profiles/addon.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    <extension point="xbmc.service" library="service.py" />
+    <extension point="xbmc.addon.metadata">
+        <news>
+v.1.4.2~beta5
+changing in logging to accomodate new logging requirements for Kodi 19
+        </news>
+        <assets>
+            <icon>icon.png</icon>
+        </assets>
+        <summary lang="en">Save audio profile and easily switch between them.</summary>
```
Comment (enen92) [inline script.audio.profiles/addon.xml:17]: en_GB, it_IT, etc

File: script.audio.profiles/resources/lib/apmonitor.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            notify.logDebug('[MONITOR] Last loaded profile: %s To switch profile: %s' % (lastProfile, profile))
+            if (lastProfile != profile and susppend_auto_change is not True) or forceload.lower() == 'true':
+                xbmc.executebuiltin('RunScript(%s, %s)' % (ADDON_ID, profile))
+            else:
+                notify.logDebug('[MONITOR] Switching omitted (same profile) or switching is susspend')
+
+
+    def getLastProfile(self):
+        try:
+            f = xbmcvfs.File(os.path.join(ADDON_PATH_DATA, 'profile'))
```
Comment (enen92) [inline script.audio.profiles/resources/lib/apmonitor.py:114]: use `with`

File: script.audio.profiles/resources/lib/profiles.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+# *  original Audio Profiles code by Regss
+# *  updates and additions through v1.4.1 by notoco and CtrlGy
+# *  updates and additions since v1.4.2 by pkscout
+
+from kodi_six import xbmc, xbmcaddon, xbmcvfs
+import json, os, sys
+import resources.lib.dialog as dialog
+import resources.lib.notify as notify
+
+ADDON = xbmcaddon.Addon()
```
Comment (enen92) [inline script.audio.profiles/resources/lib/profiles.py:13]: all these variables are used everywhere, maybe a good idea to move it to a common file where you can just import ? or define them in __all__

File: script.audio.profiles/resources/lib/profiles.py [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+        # prepare JSON string to save to file
+        if xbmc_version > 18:
+            settingsToSave = convert(settingsToSave)
+        jsonToWrite = json.dumps(settingsToSave)
+        # create dir in addon data if not exist
+        if not xbmcvfs.exists(ADDON_PATH_DATA):
+            xbmcvfs.mkdir(ADDON_PATH_DATA)
+        # save profile file
+        notify.logInfo('[SAVING SETTING]: %s' % sName[button])
+        f = xbmcvfs.File(os.path.join(ADDON_PATH_DATA, 'profile%s.json' % str(button)), 'w')
```
Comment (enen92) [inline script.audio.profiles/resources/lib/profiles.py:149]: with

File: script.audio.profiles/resources/lib/profiles.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+            # set setting
+            if 'true' in sVolume and setName == 'volume':
+                xbmc.executeJSONRPC(
+                    '{"jsonrpc": "2.0", "method": "Application.SetVolume", "params": {"volume": %s}, "id": 1}' % jsonResult['volume'])
+            else:
+                xbmc.executeJSONRPC(
+                    '{"jsonrpc": "2.0", "method": "Settings.SetSettingValue", "params": {"setting": "%s", "value": %s}, "id": 1}' % (setName, setValue))
+        notify.popup(sName[int(profile)])
+        # write curent profile
+        f = xbmcvfs.File(os.path.join(ADDON_PATH_DATA, 'profile'), 'w')
```
Comment (enen92) [inline script.audio.profiles/resources/lib/profiles.py:246]: with

File: script.audio.profiles/addon.xml (hunk 1 above)
Comment (pkscout) [reply script.audio.profiles/addon.xml:2]: Crap. I didn't push the final merge from my desktop to Github before downloading the zip. I've fixed that, but there will be a few other new lines of code to check when push the update.

File: script.audio.profiles/resources/lib/apmonitor.py (hunk 3 above)
Comment (pkscout) [reply script.audio.profiles/resources/lib/apmonitor.py:114]: When I try to change to with statements using xbmcvfs.File I get AttributeError: __exit__.  I see a pull request for xmbcvfs.File to support the exit context at: https://github.com/xbmc/xbmc/pull/16463 but Kodi 18.6 on Mac doesn't seem to have that support yet. Am I missing something?

Comment (enen92) [reply script.audio.profiles/resources/lib/apmonitor.py:114]: nvm, that's a matrix only feature

**Comment by pkscout** [comment]
OK, I think that'll do it. I feel like I may have set a record for the number of supplemental commits to get something through review though. ;-)

**Pull request merged**